pkg/controller/baremetalasset/baremetalasset_controller.go

```diff
@@ -303,9 +306,113 @@ func (r *ReconcileBareMetalAsset) Reconcile(request reconcile.Request) (reconcil
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileBareMetalAsset) checkHiveSyncSetInstance(bma *midasv1alpha1.BareMetalAsset, reqLogger logr.Logger) error {
+	found := &hivev1.SyncSetInstanceList{}
+	err := r.client.List(context.TODO(), found, client.MatchingLabels{hiveconstants.SyncSetNameLabel: bma.Name})
+
+	if err != nil {
+		reqLogger.Error(err, "Problem getting Hive SyncSetInstanceList")
+		return err
+	}
+	switch len(found.Items) {
+	case 0:
+		err = fmt.Errorf("No SyncSetInstances with label name %v and label value %v found", hiveconstants.SyncSetNameLabel, bma.Name)
+		conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+			Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+			Status:  corev1.ConditionFalse,
+			Reason:  "SyncSetInstanceNotFound",
+			Message: err.Error(),
+		})
+		return err
+	case 1:
+		resourceCount := len(found.Items[0].Status.Resources)
+		if resourceCount != 1 {
+			err = fmt.Errorf("Unexpected number of BareMetalHost resource found on SyncSetInstance status. Expected (%v) Found (%v)", 1, resourceCount)
+			conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+				Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+				Status:  corev1.ConditionFalse,
+				Reason:  "ExpectedBareMetalHostResourceCountNotFound",
+				Message: err.Error(),
+			})
+			return err
+		}
+		res := found.Items[0].Status.Resources[0]
+		if res.APIVersion != metal3v1alpha1.SchemeGroupVersion.String() || res.Kind != BareMetalHostKind {
+			err = fmt.Errorf("Unexpected resource found in SyncSetInstance status. Expected (Kind: %v APIVersion: %v) Found (Kind: %v APIVersion: %v)", BareMetalHostKind, metal3v1alpha1.SchemeGroupVersion.String(), res.Kind, res.APIVersion)
+			conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+				Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+				Status:  corev1.ConditionFalse,
+				Reason:  "ExpectedBareMetalHostResourceFound",
```

Comment:
shortened, and **not** found :smile: 
```suggestion
				Reason:  "BareMetalHostResourceNotFound",
```